File: 14-longest-common-prefix/14-longest-common-prefix.java
class Solution {
    public String longestCommonPrefix(String[] strs) {
        int idx = 0;
        int min_length = Integer.MAX_VALUE;
        for(int i=0;i<strs.length;i++){
            min_length = Math.min(min_length,strs[i].length());
        }
        
        for(int i=0;i<min_length;i++){
            for(int j=1;j<strs.length;j++){
                if(strs[j].charAt(i) != strs[j-1].charAt(i)){
                    return strs[j].substring(0,i);
                }
            }
        }
        return strs[0].substring(0,min_length);
    }
}

File: 14-longest-common-prefix/14-longest-common-prefix.cpp
class Solution {
public:
    int combine(string s1,string s2,int len) {
        int i = 0;
        while(i < len && i<s1.length() && i < s2.length() && s1[i] == s2[i]) {
            i++;
        }
        return i;
    }
    string longestCommonPrefix(vector<string>& strs) {
        int len = strs[0].length();
        
        for(int i=1;i<strs.size();i++) {
            len = min(len,combine(strs[i],strs[i-1],len));
        }
        return strs[0].substr(0,len);
    }
};